### PR TITLE
Banktransfer iban information

### PR DESCRIPTION
**Pull request by tiutiunnykleasoft**
Do not forget to clear cache

### CODE REVIEW

File: src/Service/Gateway.php [hunk 1]
```diff
@@ -92,10 +93,22 @@ public function pay(
...
                 $this->transactionStateHandler->fail($transaction->getOrderTransaction()->getId(), $salesChannelContext->getContext());
                 throw new EmsPluginException(current($order['transactions'])['reason']);
             }
+
+            /**
+             * Redirect for bank-transfer payment method
+             */
+            if (isset($order['transactions']) && current($order['transactions'])['payment_method'] == 'bank-transfer' && current($order['transactions'])['status']){
+                return new RedirectResponse(
+                implode('&',[$order['return_url'],implode('=',['order_id',$order['id']]),implode('=',['project_id',$order['project_id']])])
```
Comment (volgustleasoft) [inline src/Service/Gateway.php]: The code is difficult to read. Why not use the usual cancotation of several parts without using the implode function?

File: src/Service/Helper.php [hunk 2]
```diff
@@ -324,10 +324,33 @@ public function saveEMSLog($msg, $context) {
...
+    /**
+     *  Save the IBAN information for Bank-Transfer payment metohd
+     * @param $orderTransactionId
+     * @param $payment_method_details
+     * @param $orderRepository
+     * @param $context
+     * @return mixed
+     */
+
+    public function saveIbanInfo($orderTransactionId,$payment_method_details,$orderRepository,$context){
```
Comment (volgustleasoft) [inline src/Service/Helper.php]: How about combining functions saveIbanInfo and saveGingerOrderId into one?

File: src/Service/Gateway.php [hunk 3]
```diff
@@ -118,15 +133,26 @@ public function finalize(
...
+
+        if (isset($order['transactions']) && current($order['transactions'])['payment_method'] == 'bank-transfer'){
+            $payment_details = current($order['transactions'])['payment_method_details'];
+            $this->helper->saveIbanInfo(
+            $transaction->getOrderTransaction()->getId(),
+            $payment_details,
+            $this->orderRepository,
+            $context
+            );
+        } else
```
Comment (volgustleasoft) [inline src/Service/Gateway.php]: How about getting rid of the else condition and saving it anyway?

**Review by volgustleasoft — COMMENTED** [review]
Does ShopWare 6 support PHP version below 7?

File: src/Service/Gateway.php (hunk 1 above)
Comment (tiutiunnykleasoft) [reply src/Service/Gateway.php]: Will don

File: src/Service/Gateway.php (hunk 3 above)
Comment (tiutiunnykleasoft) [reply src/Service/Gateway.php]: I use a check for the existence of such a key, implying that only bank-transfer has it

File: src/Service/Helper.php (hunk 2 above)
Comment (tiutiunnykleasoft) [reply src/Service/Helper.php]: Was done

**Review by volgustleasoft — COMMENTED** [review]
It looks unnecessary: https://github.com/emspay/ems-online-shopware-6/blob/banktransfer_iban_information/src/Service/Gateway.php#L46

**Pull request merged**